src/herbpy/action/rogue.py

```diff
@@ -0,0 +1,321 @@
+import logging, openravepy, prpy 
+from prpy.action import ActionMethod
+from prpy.planning.base import PlanningError
+from contextlib import contextmanager 
+from prpy.util import FindCatkinResource
+import numpy, cPickle, time, os.path
+
+logger = logging.getLogger('herbpy')
+
+@ActionMethod
+def Point(robot, focus, manip=None, render=False):
+    """
+    @param robot The robot performing the point
+    @param focus The 3-D coordinate in space or object 
+                 that is being pointed at
+    @param manip The manipulator to perform the point with. 
+                 This must be the right arm
+    @param render Render tsr samples during planning
+    """
+
+    #Pointing at an object
+    if type(focus) == openravepy.openravepy_int.KinBody:
+        focus_trans = focus.GetTransform()
+        goal_name = focus.GetName()
+    #Pointing at a point in space
+    elif (type(focus) == numpy.ndarray) and (focus.ndim == 1):
+        if len(focus) != 3:
+            raise prpy.exceptions.PrPyExceptions('A point in space must \
+                    contain exactly three coordinates')
+        focus_trans = numpy.eye(4)
+        focus_trans[0:3, 3] = focus
+        goal_name = None
+    else:
+        raise prpy.exceptions.PrPyException('Focus of the point is an \
+                unknown object')
```

Comment:
This type inference is un-Pythonic and makes me nervous. For example, the "pointing at a point in space" logic will break if the user passes a `list`, `tuple`, or any other type of collection that is not an `ndarray`. This violates Python's affinity for [duck typing](https://en.wikipedia.org/wiki/Duck_typing).

I would prefer if you restructured this slightly:

1. Make a `Point` action that takes `focus_trans` and (maybe, see above) a `goal_name`.
2. Add `PointAtKinBody` and `PointAtPoint` functions for, respectively points and `KinBody`'s
3. If you think it is necessary, write a generic `PointAt` function that infers which `PointAtX` function to call.

Also, you should [always use `instanceof`, not `type` equality checks](http://legacy.python.org/dev/peps/pep-0008/#programming-recommendations) to compare types.